src/components/CountryPage/CountryPage.jsx

```diff
@@ -0,0 +1,122 @@
+import { useState, useEffect, useContext } from "react";
+import { useParams } from "react-router-dom";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faArrowLeft } from "@fortawesome/free-solid-svg-icons";
+import {
+  CountryPageWrapper,
+  StyledLink,
+  CountryInformationContainer,
+  FlagContainer,
+  NameAndDetails,
+  Details,
+} from "./CountryPage.styles";
+import addCommas from "../../utils/addCommas";
+import { DarkModeContext } from "../../Contexts/DarkModeContext";
+import BorderCountries from "../BorderCountries";
+
+export default function CountryPage() {
+  const { isDarkMode } = useContext(DarkModeContext);
+  const [country, setCountry] = useState();
+  const params = useParams();
+  const checkLocalStorage = () => {
+    const localStorageItem = localStorage.getItem(`${params.countryCode}`);
+    if (localStorageItem) {
+      setCountry(JSON.parse(localStorageItem));
+      return true;
+    }
+    return false;
+  };
+  const fetchSpecificCountry = async () => {
+    const response = await fetch(
+      `https://restcountries.com/v3.1/alpha/${params.countryCode}`
+    );
+    if (response.ok) {
+      const data = await response.json();
+      setCountry(data);
+      localStorage.setItem(`${params.countryCode}`, JSON.stringify(data));
+    }
+  };
+  useEffect(() => {
+    if (!checkLocalStorage()) {
+      fetchSpecificCountry();
+    }
+  }, [params.countryCode]);
+
+  return (
+    <CountryPageWrapper dark={isDarkMode}>
+      <StyledLink dark={isDarkMode} to="/">
+        <FontAwesomeIcon icon={faArrowLeft} />
+        Back
+      </StyledLink>
+      {country && (
+        <>
+          <CountryInformationContainer>
+            <FlagContainer>
+              <img
+                src={country[0].flags.png}
+                alt={`${country[0].name.common} flag`}
+              />
+            </FlagContainer>
+            <NameAndDetails>
+              <h1>{country[0].name.common}</h1>
```

Comment:
całkiem sporo jest tego kodu z detalami - możesz stworzyć jakiś nowy komponent na to, np CountryDetails i tam przenieść te rzeczy